Simplenote/TagListViewController.m

```diff
@@ -473,30 +481,33 @@ - (void)menuDidClose:(NSMenu *)menu
 
 - (void)controlTextDidBeginEditing:(NSNotification *)notification
 {
-    NSTextView *textView = [notification.userInfo objectForKey:@"NSFieldEditor"];
-    self.tagNameBeingEdited = [textView.string copy];
+    self.tagNameBeingEdited = [notification.fieldEditor.string copy];
 }
 
 - (void)controlTextDidEndEditing:(NSNotification *)notification
 {
-    if (self.tagNameBeingEdited) {
-        // This can get triggered before renaming has started; don't do anything in that case
+    // This can get triggered before renaming has started; don't do anything in that case
+    if (!self.tagNameBeingEdited) {
+        return;
+    }
 
-        NSTextView *textView = [notification.userInfo objectForKey:@"NSFieldEditor"];
-        
-        [textView setSelectedRange:NSMakeRange(0, 0)]; // force de-selection of text
-        
-        // Note:
-        // Send a *COPY* of the string. Otherwise the internal string will be exposed, and this may lead to
-        // weird side effects.
-        NSString *newTagName = [textView.string copy];
-        
-        BOOL tagAlreadyExists = [self tagWithName:newTagName] != nil;
-        if ([newTagName length] > 0 && !tagAlreadyExists && ![self.tagNameBeingEdited isEqualToString:newTagName])
-            [self changeTagName:self.tagNameBeingEdited toName:newTagName];
-        
-        self.tagNameBeingEdited = nil;
+    // Force de-selection of text
+    NSTextView *textView = notification.fieldEditor;
+    [textView setSelectedRange:NSMakeRange(0, 0)];
+
+    // Send a *COPY* of the string. Otherwise the internal string will be exposed, and this may lead to
+    // weird side effects.
+    NSString *newTagName    = [textView.string copy];
+    BOOL newTagNotFound     = [self tagWithName:newTagName] == nil;
```

Comment:
this feels like we're in the rename logic. if that's the case we should note that it's possible for the new tag to exist because of the tag-hash collapsing.

it's legitimate to change the casing of a tag: rename `markdown` to `Markdown`

in those cases `[self tagWithName:@"Markdown"]` will return the tag with the name `markdown`

to get around this we probably need to check not if the new tag exists but if it's different than the existing one

```objective-c
Tag *oldTag = [self tagWithName:self.tagNameBeingEdited];
Tag *newTag = [self tagWithName:newTagName];
BOOL isProperRename = newTag == nil || newTag == oldTag;

if (oldTagWasChanged && isProperRename && newTagName.length > 0) {
	…
```

Reply:
TOTALLY MISSED that scenario!!! ty!!